vendor/curl/curl.odin

```diff
@@ -0,0 +1,362 @@
+// Package curl implements minimal bindings to libcurl.
+//
+// This implementation is based on curl.h version 8.11.1:
+// [[ More; https://github.com/curl/curl/blob/curl-8_11_1/include/curl/curl.h ]]
+package curl
+
+import "base:runtime"
+import "core:c"
+import "core:strings"
+
+// CURL represents an easy handle.
+//
+// **Note**: An easy handle can perform one transfer at a time but can be reused
+// for multiple transfers.
+CURL :: distinct rawptr
+
+// CURL_GLOBAL_ALL combines all initialization flags.
+CURL_GLOBAL_ALL: i64 = 3
+
+// CURLOPT represents options for a transfer.
+CURLOPT :: enum i32 {
+  // Basic options
+  WRITEFUNCTION  = 20011, // Callback for receiving data
+  WRITEDATA      = 10001, // Data pointer passed to write callback
+  URL            = 10002, // URL to request
+  USERAGENT      = 10018, // User-Agent header
+  HTTPHEADER     = 10023, // Custom HTTP headers
+  POST           = 47,    // Enable POST
+  POSTFIELDS     = 10015, // POST data
+  VERBOSE        = 41,    // Enable verbose output
+  TIMEOUT_MS     = 155,   // Timeout in milliseconds
+  FOLLOWLOCATION = 52,    // Follow redirects
+  MAXREDIRS      = 68,    // Maximum redirects to follow
+
+  // SSL/TLS options
+  SSL_VERIFYPEER    = 64,    // Verify the peer (0 or 1)
+  SSL_VERIFYHOST    = 81,    // Host verification level (0, 1, or 2)
+  SSL_VERIFYSTATUS  = 232,   // Check certificate status
+  SSLVERSION        = 32,    // SSL protocol version
+  SSL_CIPHER_LIST   = 10083, // Allowed ciphers
+
+  // SSL certificate options
+  CAINFO            = 10065, // Path to CA bundle file
+  CAPATH            = 10097, // Directory of CA certificates
+  SSLCERT           = 10025, // Client certificate path
+  SSLKEY            = 10087, // Private key file path
+  KEYPASSWD         = 10026, // Private key password
+
+  // SSL engine options
+  SSLENGINE         = 10089, // Crypto engine to use
+  SSLENGINE_DEFAULT = 90,    // Set default crypto engine
+}
+
+// CURLINFO represents information retrieval options.
+CURLINFO :: enum i32 {
+  // Response status code
+  Response_Code = 0x200000 + 2,
+}
+
+// Error represents libcurl error codes.
+//
+// This maps directly to CURLcode error values from curl.h.
+Error :: enum {
```

Comment:
I know this wraps the other calls, but why not keep it the same? e.g. `enum c.int`?